main/tests/server/src/com/google/refine/util/HttpClientTests.java

```diff
@@ -27,4 +32,12 @@ public void fromHostsToPattern() {
         Assert.assertTrue(pattern3.matcher("random.domain.any.com").matches());
         Assert.assertTrue(pattern3.matcher("myhosts.mydomain.mine").matches());
     }
+
+    @Test
+    public void urlEscaping() throws MalformedURLException, URISyntaxException {
+        final String ARABIC_PATH = "احمد";
+        final String BASE_URL = "https://example.com/";
+        final String TEST_URL = BASE_URL + ARABIC_PATH;
+        assertEquals(HttpClient.getEscapedUrl(TEST_URL), BASE_URL + UrlEscapers.urlPathSegmentEscaper().escape(ARABIC_PATH));
+    }
```

Comment:
Would it not be worth adding some test cases to demonstrate the handling of already-escaped URLs, to make sure they are not doubly escaped?